File: javascript/3264.Final-Array-State-After-K-Multiplication-Operations-I.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @param {number} multiplier
 * @return {number[]}
 */
var getFinalState = function (nums, k, multiplier) {
  const numsCopy = [...nums];

  const getMinIndex = () => {
    let min = 0;

    for (let i = 0; i < numsCopy.length; i++) {
      if (numsCopy[i] < numsCopy[min]) {
        min = i;
      }
    }

    return min;
  };

  while (k) {
    const idx = getMinIndex();
    numsCopy[idx] *= multiplier;
    k -= 1;
  }

  return numsCopy;
};

File: python/3264.Final-Array-State-After-K-Multiplication-Operations-I.py
class Solution:
    def getFinalState(self, nums: List[int], k: int, multiplier: int) -> List[int]:
        heap = [(val, idx) for idx, val in enumerate(nums)]
        heapq.heapify(heap)

        nums_copy = nums[:]

        while k:
            val, idx = heapq.heappop(heap)
            new_val = val * multiplier
            nums_copy[idx] = new_val
            heapq.heappush(heap, (new_val, idx))
            k -= 1

        return nums_copy